pkg/mimir/modules.go

```diff
@@ -1034,6 +1037,23 @@ func (t *Mimir) initUsageStats() (services.Service, error) {
 	return t.UsageStatsReporter, nil
 }
 
+func (t *Mimir) initContinuousTest() (services.Service, error) {
+	start := func(ctx context.Context) error {
+		client, err := continuoustest.NewClient(t.Cfg.ContinuousTest.Client, util_log.Logger)
+		if err != nil {
+			return errors.Wrap(err, "Failed to initialize continuous-test client")
+		}
+
+		t.ContinuousTestManager = continuoustest.NewManager(t.Cfg.ContinuousTest.Manager, util_log.Logger)
+		t.ContinuousTestManager.AddTest(continuoustest.NewWriteReadSeriesTest(t.Cfg.ContinuousTest.WriteReadSeriesTest, client, util_log.Logger, t.Registerer))
+		return nil
+	}
+
+	return services.NewBasicService(start, func(ctx context.Context) error {
```

Comment:
Usually the way you'd use the `service` stuff is to have `continoustest.Manager` implement the `service.Service` interface and return an instance of it from this method. Then the module system handles calling start and run methods.

Reply:
True! That was the original design. I don't know how I got here from there, but happy to go back to it as that's how the other modules behave.

Reply:
Nope! Nevermind. I just converted it and got the errors I was getting beforehand when using the Manager as the service. Basically yes, that is the pattern we use in many other modules but they are meant to be continuously running with different behaviors for stopping etc. whereas if you look at ActivityTracker as an example for NewIdleService, it's worth it to use that dskit infrastructure instead of writing a new AwaitRunning func etc for it to behave properly. All I want is to init, and then be able to call Run and wrapping in BasicService does that here cleanly I think. I do agree with your previous comment for cleanly running it all as init, and then just pass it the Run func.